test/spiffe_id/test_trust_domain.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+
+from src.pyspiffe.spiffe_id.trust_domain import TrustDomain
+
+
+@pytest.mark.parametrize(
+    'test_input,expected',
+    [
+        ('domain.test', 'domain.test'),
+        (' doMain.tesT  ', 'domain.test'),
+        ('spiffe://domAin.Test', 'domain.test'),
+        ('spiffe://domain.test/path/element', 'domain.test'),
+        ('spiffe://domain.test/spiffe://domain.test/path/element', 'domain.test'),
+        ('spiffe://domain.test/spiffe://domain.test:80/path/element', 'domain.test'),
```

Comment:
even if we can extract the domain name, is this actually a valid trust domain according to the SPIFFE spec?

Reply:
Yes,  the domain name is taken from the authority, we don't care about the rest, only that the authority in this case cannot have a port. The rest of the URL is validated in case you want to create a SpiffeID from it.